pkg/providers/tinkerbell/comparator.go

```diff
@@ -0,0 +1,30 @@
+package tinkerbell
+
+import (
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+)
+
+// ControlPlaneVersionChange detects kubernetes version has changed between new and old KubeadminControlPlane objects.
+func ControlPlaneVersionChange(oldKCP *controlplanev1.KubeadmControlPlane, newKCP *controlplanev1.KubeadmControlPlane) bool {
+	if oldKCP != nil && newKCP != nil {
+		return oldKCP.Spec.Version != newKCP.Spec.Version
+	}
+	return false
+}
+
+// ControlPlaneReplicasChange detects whether replica count has changed between new and old KubeadminControlPlane objects.
+func ControlPlaneReplicasChange(oldKCP *controlplanev1.KubeadmControlPlane, newKCP *controlplanev1.KubeadmControlPlane) bool {
```

Comment:
I thought you were gonna make these comparisons not dependent on the types, like implementing an interface?

Reply:
Implemented comparator interfaces for kubernetes version and replica count retrieval